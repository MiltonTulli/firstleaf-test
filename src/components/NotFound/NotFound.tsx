import React from 'react';
import { Typography, Button } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import { useHistory } from 'react-router-dom';
import R from '../../utils/routes';

const styles = makeStyles(theme => ({
  root: {
    padding: theme.spacing(4),
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    justifyContent: 'center'
  },
  text: {
    marginBottom: theme.spacing(2)
  }
}));

export const NotFound: React.FunctionComponent = () => {
  const classes = styles();
  const history = useHistory();
  return (
    <div className={classes.root}>
      <Typography className={classes.text} variant="h4">
        Ups!, Page Not Found
      </Typography>
      <Button variant="outlined" onClick={() => history.push(R.APP)}>
        Go to App
      </Button>
    </div>
  );
};
