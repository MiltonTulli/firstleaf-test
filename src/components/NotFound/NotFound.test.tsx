import React from 'react';
import renderer from 'react-test-renderer';
import { NotFound } from './NotFound';

describe('<NotFound>', () => {
  it('Should match snapshot', () => {
    const wrapper = renderer.create(<NotFound />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
