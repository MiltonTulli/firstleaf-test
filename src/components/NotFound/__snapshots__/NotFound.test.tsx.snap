// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound> Should match snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <h4
    className="MuiTypography-root makeStyles-text-2 MuiTypography-h4"
  >
    Ups!, Page Not Found
  </h4>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Go to App
    </span>
  </button>
</div>
`;
